test/jdk/java/util/concurrent/SynchronousQueue/Fairness.java

```diff
@@ -23,48 +23,52 @@
 
 /*
  * @test
- * @bug 4992438 6633113
+ * @bug 4992438 6633113 8314515
  * @summary Checks that fairness setting is respected.
  */
 
+import java.util.ArrayList;
 import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.SynchronousQueue;
-import java.util.concurrent.locks.Condition;
-import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class Fairness {
-    private static void testFairness(boolean fair,
-                                     final BlockingQueue<Integer> q)
+    private static void testFairness(boolean fair, SynchronousQueue<Integer> q)
         throws Throwable
     {
-        final ReentrantLock lock = new ReentrantLock();
-        final Condition ready = lock.newCondition();
-        final int threadCount = 10;
-        final Throwable[] badness = new Throwable[1];
-        lock.lock();
+        int threadCount = ThreadLocalRandom.current().nextInt(2, 8);
```

Comment:
@Martin-Buchholz I'm a bit weary about making the number of threads random—might make it tricky to get to reproduce if the test fails?

Reply:
Suppose the test fails only with a specific number of threads.  I would want to know that!  At least once in the past a j.u.c. bug has been found due to randomization of concurrency level in a test.  I can't understand the current industry obsession with deterministic tests, except that too many  organizations don't commit to hunting down flakes.

Here the error detail has been changed to report the threadcount, in case of failure.